# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1467354611 25200
#      Thu Jun 30 23:30:11 2016 -0700
# Node ID a49ed7b7b66fa61bbbb1a259642ffb8f17d216c6
# Parent  4eafa2cc599f214897dc7e77e4539d10e2ce8668
build: delete HG-ID on 'make distclean' and 'make maintainer-clean'

* Makefile.am (MAINTAINERCLEANFILES): Add HG-ID to the list.
(distclean-local): Delete HG-ID only if building outside of $(srcdir).

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -294,18 +294,19 @@ CLEANFILES += \
   make-vars
 
 DISTCLEANFILES += \
   $(DIRSTAMP_FILES) \
   $(GEN_CONFIG_SHELL) \
   .gdbinit
 
 MAINTAINERCLEANFILES += \
+  $(BUILT_DISTFILES) \
   ChangeLog \
-  $(BUILT_DISTFILES)
+  HG-ID
 
 CONFIG_FILES = @ac_config_headers@ @ac_config_files@
 
 octinclude_HEADERS += oct-conf-post.h
 nodist_octinclude_HEADERS += config.h octave-config.h
 
 ALL_LOCAL_TARGETS += \
   $(OCTAVE_INTERPRETER_TARGETS) \
@@ -414,16 +415,17 @@ DIRS_TO_MAKE = \
 installdirs-local:
 	$(MKDIR_P) $(addprefix $(DESTDIR), $(DIRS_TO_MAKE))
 
 install-data-local: installdirs-local
 
 clean-local: doc-clean
 
 distclean-local:
+	if [ "x${srcdir}" != "x." ]; then rm -f HG-ID; fi
 
 maintainer-clean-local: doc-maintainer-clean
 
 define move_if_change_rule
   if [ -s $(1) ]; then \
     ${SHELL} ${top_srcdir}/build-aux/move-if-change $(1) $(2); \
   else \
     echo "$(1) is empty!" 1>&2; \
